lib/sign.js

```diff
@@ -16,14 +17,36 @@ function sign (manifest, program, options, config) {
     AMOClient: DefaultAMOClient,
   }, config);
 
-  var client = new config.AMOClient({
-    apiKey: options.apiKey, apiSecret: options.apiSecret,
-    apiUrlPrefix: options.apiUrlPrefix,
-  });
+  return when.promise(function(resolve, reject) {
+
+    var missingOptions = [];
+    [
+      {value: options.apiKey, flag: '--api-key'},
+      {value: options.apiSecret, flag: '--api-secret'},
+    ]
+    .forEach(function(opt) {
```

Comment:
I think this looks really weird. I'd rather see the `forEach` on the same line as the end of the array or giving the array a name. At first glance I thought this was supposed to be assigned to `missingOptions`.

I also find it strange that commander won't handle this. Oh well.